plugin/pkg/admission/namespace/lifecycle/admission.go

```diff
@@ -45,11 +46,30 @@ func init() {
 type lifecycle struct {
 	*admission.Handler
 	client             clientset.Interface
-	store              cache.Store
 	immortalNamespaces sets.String
+	namespaceInformer  framework.SharedIndexInformer
+	// forceLiveLookupCache holds a list of entries for namespaces that we have a strong reason to believe are stale in our local cache.
+	// if a namespace is in this cache, then we will ignore our local state and always fetch latest from api server.
+	forceLiveLookupCache *lru.Cache
```

Comment:
Ideally, I would like a layer, but thought that was a good follow-up with what @hodovska is doing.

In this case, I was less concerned about resource version comparison since all I really care about is that the version in my local cache is now in terminating phase which is the check we have later in this file.  Good point on the mass deletion paging out the LRU.  Not sure how concerned I am on that scenario if we will swap out with the layered store.